libpangolin/include/pangolin/BfsMining/engine.h

```diff
@@ -1,10 +1,26 @@
 #include "galois/Galois.h"
 #include "pangolin/res_man.h"
 #include "pangolin/BfsMining/embedding_list.h"
+#include "llvm/Support/CommandLine.h"
+
+//! Flag that forces user to be aware that they should be passing in a
+//! symmetric graph.
+static cll::opt<bool> symmetricGraph(
+    "symmetricGraph",
+    cll::desc("Flag should be used to make user aware they should be passing a "
+              "symmetric graph to this program"),
+    cll::init(false));
 
 int main(int argc, char** argv) {
   galois::SharedMemSys G;
   LonestarMineStart(argc, argv, name, desc, url);
+
+  if (!symmetricGraph) {
+    GALOIS_DIE("Mining apps requires a symmetric graph input;"
+               " please use the -symmetricGraph flag "
+               " to indicate the input is a symmetric graph");
+  }
+
```

Comment:
You might want to mention that there's an assumption that the graph must be cleaned for mining applications as well.

That might require yet another "cleanGraph" flag.

Reply:
Good point. However, I'm not sure whether "clean" term is well understood. The standard term is simple graph:
https://en.wikipedia.org/wiki/Multiple_edges
So let's use "-simpleGraph" and the help message should say "Specify that the graph is simple (has no multiple edges)".

Reply:
Don't forget no self-loops as well.

Reply:
Noted